### PR TITLE
Reference as much of the VS SDK as we can via NuGet packages

### PR DESCRIPTION
**Pull request by jasonmalinowski**
This brought along some references to Microsoft.VisualStudio.Threading that we previously didn't have. Nothing was wrong with this, but it meant I had to fix some namespace qualifications that were assuming the "Threading" namespace was unambiguous.

The intent behind this change is to lock the VS version that these projects depend upon, rather than picking up whatever version you might have on your box. Thus, if you're building with Visual Studio "15" you can't accidentally pick up a 15-only dependency.

There's definitely more assemblies to go; this is just a first pass.

*Review:* @dotnet/roslyn-ide, @dotnet/roslyn-interactive, @jaredpar

### CODE REVIEW

File: src/EditorFeatures/Core/Shared/Utilities/BrushToColorConverter.cs [hunk 1]
```diff
@@ -1,39 +0,0 @@
-// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
-
-using System;
-using System.Globalization;
-using System.Windows.Data;
-using System.Windows.Media;
-
-namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
-{
-    internal class BrushToColorConverter : IValueConverter
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Shared/Utilities/BrushToColorConverter.cs]: This file was copied/pasted but is available from Microsoft.VisualStudio.PlatformUI which we now get as a NuGet package.

File: src/EditorFeatures/Core/project.json [hunk 2]
```diff
@@ -1,6 +1,10 @@
 ﻿{
   "dependencies": {
-    "System.Collections.Immutable": "1.1.37"
+    "System.Collections.Immutable": "1.1.37",
```
Comment (tmat) [inline src/EditorFeatures/Core/project.json]: Why is this referencing Immutable? Shouldn't it flow from downlevel project refs?

File: src/VisualStudio/VisualStudioDiagnosticsToolWindow/VisualStudioDiagnosticsWindow.csproj [hunk 3]
```diff
@@ -90,33 +90,13 @@
...
       <Private>false</Private>
-      <HintPath>$(DevEnvDir)\PrivateAssemblies\Microsoft.VisualStudio.Shell.14.0.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.VisualStudio.Shell.Immutable.10.0, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"><Private>false</Private></Reference>
-    <Reference Include="Microsoft.VisualStudio.Shell.Interop.10.0, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <EmbedInteropTypes>True</EmbedInteropTypes>
-    </Reference>
-    <Reference Include="Microsoft.VisualStudio.Shell.Interop.11.0, Version=11.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <EmbedInteropTypes>True</EmbedInteropTypes>
+    <Reference Include="envdte80, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
```
Comment (jaredpar) [inline src/VisualStudio/VisualStudioDiagnosticsToolWindow/VisualStudioDiagnosticsWindow.csproj]: DTE isn't in the SDK??? 

Maybe we can grab Sam's NuGet packages.  Think they have DTE.

**Comment by jaredpar** [comment]
:+1: 

Do you have the list of DLLs that we couldn't move over?

**Comment by tmat** [comment]
Nice! :+1:

File: src/EditorFeatures/Core/project.json (hunk 2 above)
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/project.json]: Fixing.

**Comment by jasonmalinowski** [comment]
@jaredpar: I'll be emailing the team with a list. I'll cc you.

**Comment by jasonmalinowski** [comment]
@dotnet-bot, test vsi please.

File: src/Compilers/Extension/CompilerExtension.csproj [hunk 4]
```diff
@@ -61,44 +61,6 @@
...
-      <Private>false</Private>
-    </Reference>
-    <Reference Include="Microsoft.VisualStudio.Shell.Interop.8.0">
-      <Private>false</Private>
-    </Reference>
-    <Reference Include="Microsoft.VisualStudio.Shell.Interop.9.0">
-      <Private>false</Private>
-    </Reference>
-    <Reference Include="Microsoft.VisualStudio.Shell.Interop.10.0">
-      <EmbedInteropTypes>True</EmbedInteropTypes>
```
Comment (KevinH-MS) [inline src/Compilers/Extension/CompilerExtension.csproj]: How is ```<EmbedInteropTypes>true</EmbedInteropTypes>``` expressed, or was it non-important/superfluous in these cases?

**Comment by KevinH-MS** [comment]
Changes LGTM, modulo VS integration test failures...

File: src/Compilers/Extension/CompilerExtension.csproj (hunk 4 above)
Comment (jasonmalinowski) [inline src/Compilers/Extension/CompilerExtension.csproj]: It's not. This either was superfluous, or perhaps in our new world we're not even referencing it. I generally did this by removing VS references and re-adding packages as I needed them if I wasn't 100% sure we actually needed the reference.

**Comment by jasonmalinowski** [comment]
@dotnet-bot, test vsi please.

File: src/Compilers/Extension/CompilerExtension.csproj (hunk 4 above)
Comment (jaredpar) [inline src/Compilers/Extension/CompilerExtension.csproj]: Is there a way to specified `<EmbedInteropTypes>` in project.json?  If not it's something we should push on.

**Comment by jasonmalinowski** [comment]
@davkean, so here was the PR I was discussing earlier. After fixing whatever I broke, of course, if I merge this in does this break anything else you're looking to do? I assume this gets you closer to what you were looking to do because at least our dependencies become NuGet references that can flow vs. random VS references on the machine that come from who knows where.

**Comment by jasonmalinowski** [comment]
@dotnet-bot, retest vsi please. Those tests failed due to a break in internal that was fixed by @jmarolf.

File: src/Compilers/Extension/CompilerExtension.csproj (hunk 4 above)
Comment (jasonmalinowski) [inline src/Compilers/Extension/CompilerExtension.csproj]: Turns out this was actually very important. I've updated some targets to work around this issue that were lovingly inspired (i.e. copied and tweaked) from the SDK NuGet package to work around this issue.

**Comment by jasonmalinowski** [comment]
@dotnet-bot, test vsi please.

**Comment by jasonmalinowski** [comment]
@dotnet-bot, test vsi please.

**Pull request merged**